test/acceptance/tests/ingress-gateway/ingress_gateway_test.go

```diff
@@ -0,0 +1,192 @@
+package connect
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	"github.com/hashicorp/consul-helm/test/acceptance/framework"
+	"github.com/hashicorp/consul-helm/test/acceptance/helpers"
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/sdk/testutil/retry"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// Test that ingress gateways work in a default installation.
+func TestIngressGatewayDefault(t *testing.T) {
+	env := suite.Environment()
+
+	// Initially we start without ingress gateways enabled because we need
+	// to create their config entry first.
```

Comment:
Is this a requirement? I feel like I was able to do that with a single install, but maybe I'm missing something. Also, I don't think we mention it in [our docs](https://www.consul.io/docs/k8s/connect/ingress-gateways).

Reply:
Fixed to remove the upgrade. I left in the code for upgrade because it might still be useful.